plugins/right-sizer/src/controller/controller.go

```diff
@@ -0,0 +1,264 @@
+		if relatedReportItems != nil {
+			eventSummary := fmt.Sprintf("%s %s/%s %s", event.Reason, event.InvolvedObject.Kind, event.InvolvedObject.Namespace, event.InvolvedObject.Name)
+			glog.V(1).Infof("going to remove report items related to event %q: %v", eventSummary, relatedReportItems)
+			if c.reportBuilder.RemoveItems(*relatedReportItems) {
+				c.reportBuilder.WriteConfigMap()
+			}
+		}
+		return
+	}
+	pod, err := c.podLister.Pods(event.InvolvedObject.Namespace).Get(event.InvolvedObject.Name)
+	if err != nil {
+		glog.Errorf("Failed to retrieve pod %s/%s, due to: %v", event.InvolvedObject.Namespace, event.InvolvedObject.Name, err)
+		return
+	}
+	c.evaluatePodStatus(pod)
+}
+
+func (c *Controller) evaluateEventUpdate(eventUpdate *eventUpdateGroup) {
+	event := eventUpdate.newEvent
+	if eventUpdate.oldEvent == nil {
+		glog.V(4).Infof("No old event present for event %s/%s (count: %d), reason: %s, involved object: %s, skipping processing", event.ObjectMeta.Namespace, event.ObjectMeta.Name, event.Count, event.Reason, event.InvolvedObject.Kind)
+		return
+	}
+	if reflect.DeepEqual(eventUpdate.oldEvent, eventUpdate.newEvent) {
+		glog.V(4).Infof("Event %s/%s (count: %d), reason: %s, involved object: %s, did not change: skipping processing", event.ObjectMeta.Namespace, event.ObjectMeta.Name, event.Count, event.Reason, event.InvolvedObject.Kind)
+		return
+	}
+	if isSameEventOccurrence(eventUpdate) {
+		glog.V(3).Infof("Event %s/%s (count: %d), reason: %s, involved object: %s, did not change wrt. to restart count: skipping processing", eventUpdate.newEvent.ObjectMeta.Namespace, eventUpdate.newEvent.ObjectMeta.Name, eventUpdate.newEvent.Count, eventUpdate.newEvent.Reason, eventUpdate.newEvent.InvolvedObject.Kind)
+		return
+	}
+	if !isContainerStartedEvent(event) {
+		// IF this update matches a kind/namespace/name of a pod-controller in the
+		// report, remove related report items.
+		relatedReportItems := c.reportBuilder.MatchItemsWithOlderResourceVersion(event.InvolvedObject.ResourceVersion, event.InvolvedObject.Kind, event.InvolvedObject.Namespace, event.InvolvedObject.Name)
+		if relatedReportItems != nil {
+			eventSummary := fmt.Sprintf("%s %s/%s %s", event.Reason, event.InvolvedObject.Kind, event.InvolvedObject.Namespace, event.InvolvedObject.Name)
+			glog.V(1).Infof("going to remove report items related to event %q: %v", eventSummary, relatedReportItems)
+			if c.reportBuilder.RemoveItems(*relatedReportItems) {
+				c.reportBuilder.WriteConfigMap()
+			}
+		}
+		return
+	}
+	pod, err := c.podLister.Pods(event.InvolvedObject.Namespace).Get(event.InvolvedObject.Name)
+	if err != nil {
+		glog.Errorf("Failed to retrieve pod %s/%s, due to: %v", event.InvolvedObject.Namespace, event.InvolvedObject.Name, err)
+		return
+	}
+	c.evaluatePodStatus(pod)
```

Comment:
the logic above was duplicated in `evaluateEvent` and here ... makes sense to refactor it